os/fs/driver/mtd/smart.c

```diff
@@ -521,6 +525,28 @@ static const struct block_operations g_bops = {
  ****************************************************************************/
 
 #ifdef CONFIG_DEBUG_FS
+#ifndef NXFUSE_HOST_BUILD
+static int smart_dumpext(FAR struct smart_struct_s *dev)
+{
+	int ret = OK;
+	int i;
+	int j;
+
+	for (i = 0; i < (dev->geo.neraseblocks * dev->sectorsPerBlk); i++) {
+		ret = MTD_BREAD(dev->mtd, i * dev->mtdBlksPerSector, dev->mtdBlksPerSector, (FAR uint8_t *)dev->rwbuffer);
+		if (ret != dev->mtdBlksPerSector) {
+			return -EIO;
+		}
+
+		for (j = 0; j < dev->sectorsize; j++) {
```

Comment:
Too deep dependency here. It should work stand alone.
smart should return buffer only, then pro will get the buffer filled with data.
then dumper should write through the up_lowputc.

Reply:
I was considering the same but had 1 issue
Getting only "n" bytes at a time in the buffer, we have to request many many times to get complete partition data (2Mb or more)
So it will take a lot of time.

Also, the buffer memory will have to be allocated after the crash, so is that safe?